test/Common/TestHelpers.cs

```diff
@@ -196,9 +200,48 @@ public static DurableTaskOptions GetDurableTaskOptionsForStorageProvider(string
             }
         }
 
+        // Create a valid task hub from the test name, and add a random suffix to avoid conflicts
         public static string GetTaskHubNameFromTestName(string testName, bool enableExtendedSessions)
         {
-            return testName.Replace("_", "") + (enableExtendedSessions ? "EX" : "") + PlatformSpecificHelpers.VersionSuffix;
+
+            string strippedTestName = testName.Replace("_", "");
+            string truncatedTestName = strippedTestName.Substring(0, Math.Min(35, strippedTestName.Length));
+            string deterministicSuffix = (enableExtendedSessions ? "EX" : "") + PlatformSpecificHelpers.VersionSuffix;
+            string randomSuffix = Guid.NewGuid().ToString().Substring(0, 4);
+            return truncatedTestName + deterministicSuffix + randomSuffix;
```

Comment:
A few thoughts here:
(1) I'm a tiny bit concerned about the truncation step here. Is it necessary? I could imagine equivalent truncated names
(2) The  `deterministicSuffix` seems to be quite different from the `randomSuffix` in terms of their purpose, so I would prefer we renamed `deterministicSuffix` to `extendedSessionsSuffix` or something like that.

Reply:
1) Equivalent truncated names don't matter too much from a functionality standpoint, as the random suffix will cause the task hubs to be unique. The main downside is debugging failures by looking at the storage artifacts for a taskhub if there is a conflict. It will definitely be a bit more annoying to find which task hub corresponds to your test if there are two tests with equivalent names, but we need to do the truncation because taskhubs have to be less than ~50 characters due to storage naming limits.

2) `extendedSessionsSuffix` doesn't cover all that the suffix contains, because it also contains the Functions Version that we ran the test on. Maybe a `testPropertiesSuffix` would work, but I don't think that is necessarily any clearer than deterministic suffix. Essentially we are tagging on two suffixes, one which is deterministic for the test, and another that adds randomness.

Reply:
Thanks for clarifying. I won't block on this but I would personally prefer the `testPropertiesSuffix` name, as I think it provides more context :)